### ISSUE TITLE
Is there an analogue of the function 'softmax_cross_entropy_with_logits' from TensorFlow?

### ISSUE THREAD

**Issue by iliagrigorevdev** [mlpack/mlpack#2682]
Is there an analogue of the function 'softmax_cross_entropy_with_logits' (https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits) from TensorFlow?

**Comment by zoq** [comment]
Hello @iliagrigorevdev, you could use Softmax (https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/softmax.hpp) as the last layer and CrossEntropyError(https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/loss_functions/cross_entropy_error.hpp) as the loss function. Does this make sense?

**Comment by iliagrigorevdev** [comment]
Yes it makes sense. Thanks, @zoq! If I understand it right, this loss from TensorFlow is equal to '-sum(y*log(softmax(x)))' and in mlpack it will be '-sum(y*log(softmax(x)) + (1-y)*log(1 - softmax(x))))' (according to the source code 'mlpack-3.4.1/src/mlpack/methods/ann/loss_functions/cross_entropy_error_impl.hpp'). I am wondering if the result will be the same?

**Comment by zoq** [comment]
You are right, this will not return the same result, unless I missed something. Maybe easier to add `softmax_cross_entropy_with_logits` as a new function.
